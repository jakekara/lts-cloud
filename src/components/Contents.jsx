import React from 'react';
import Hero from './Hero';

const Contents = props => (
  <div>
    <Hero text='Table of Contents' />
    <div className='container'>
      <h2>The Holocaust Testimonials of the Name Collection</h2>
      <Table />
    </div>
  </div>
)

const Table = props => (
  <div className='toc-table'>
    {items.map((i, idx) => (
      <div key={idx} className='toc-row'>
        <span className='number'>{idx + 1}.</span>
        <div className='text'>
          <div>{ 'Oral history interview with ' + i }</div>
          <div className='meta'>Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.</div>
        </div>
      </div>
    ))}
  </div>
)

const items = [
  'Mario Speedwagon',
  'Petey Cruiser',
  'Anna Sthesia',
  'Paul Molive',
  'Anna Mull',
  'Gail Forcewind',
  'Paige Turner',
  'Bob Frapples',
]

export default Contents;