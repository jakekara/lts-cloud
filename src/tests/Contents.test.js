import React from 'react';
import Contents from '../components/Contents';
import { createBrowserHistory } from 'history'
import { Router } from 'react-router';
import renderer from 'react-test-renderer';

it('renders', () => {
  const tree = renderer
    .create(
      <Router history={createBrowserHistory()}>
        <Contents/>
      </Router>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});