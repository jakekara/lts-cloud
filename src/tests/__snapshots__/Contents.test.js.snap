// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    className="hero background-image"
  >
    <h1>
      Table of Contents
    </h1>
  </div>
  <div
    className="container"
  >
    <h2>
      The Holocaust Testimonials of the Name Collection
    </h2>
    <div
      className="toc-table"
    >
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          1
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Mario Speedwagon
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          2
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Petey Cruiser
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          3
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Anna Sthesia
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          4
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Paul Molive
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          5
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Anna Mull
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          6
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Gail Forcewind
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          7
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Paige Turner
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
      <div
        className="toc-row"
      >
        <span
          className="number"
        >
          8
          .
        </span>
        <div
          className="text"
        >
          <div>
            Oral history interview with Bob Frapples
          </div>
          <div
            className="meta"
          >
            Accession Number: 1234. Courtesy of the Fortunoff Archive. Other metadata as needed.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
